exporter/otlphttpexporter/otlp.go

```diff
@@ -178,15 +178,44 @@ func (e *exporter) export(ctx context.Context, url string, request []byte) error
 		return exporterhelper.NewThrottleRetry(formattedErr, time.Duration(retryAfter)*time.Second)
 	}
 
-	if resp.StatusCode == http.StatusBadRequest {
-		// Report the failure as permanent if the server thinks the request is malformed.
+	if isPermanentClientFailure(resp.StatusCode) {
+		// Do not retry; report the failure as permanent if the server thinks the request is malformed.
 		return consumererror.NewPermanent(formattedErr)
 	}
 
 	// All other errors are retryable, so don't wrap them in consumererror.NewPermanent().
 	return formattedErr
 }
 
+func isPermanentClientFailure(code int) bool {
+	// 400 - bad request
+	if code == http.StatusBadRequest {
+		return true
+	}
+
+	// 402 - payment required typically means that an auth token isn't valid anymore and as such, we deem it as permanent
+	if code == http.StatusPaymentRequired {
+		return true
+	}
+
+	// 413 - request size is too large
+	if code == http.StatusRequestEntityTooLarge {
+		return true
+	}
+
+	// 414 - uri is too long
+	if code == http.StatusRequestURITooLong {
+		return true
+	}
+
+	// 431 - headers too large
+	if code == http.StatusRequestHeaderFieldsTooLarge {
+		return true
+	}
+
+	return false
```

Comment:
Use a switch statement?

Reply:
The key original purpose of such structure is for code commenting.
Sure @bogdandrutu , I can make the change.     switch statement looks similar though:
```
	switch code {
            case http.StatusBadRequest:
		// 400 - bad request
		isPermanentError = true
            case http.StatusPaymentRequired:
		// 402 - payment required typically means that an auth token isn't valid anymore ...  
		isPermanentError = true
	    case http.StatusRequestEntityTooLarge:
		// 413 - request size is too large
		isPermanentError = true
	    case http.StatusRequestURITooLong:
		// 414 - uri is too long
		isPermanentError = true
	    case http.StatusRequestHeaderFieldsTooLarge:
		// 431 - headers too large
		isPermanentError = true
	    default:
		isPermanentError = false
	}

```